subscription/src/main/java/org/killbill/billing/subscription/api/transfer/DefaultSubscriptionBaseTransferApi.java

```diff
@@ -114,66 +116,76 @@ private SubscriptionBaseEvent createEvent(final boolean firstEvent, final Existi
                            PhaseEventData.createNextPhaseEvent(subscription.getId(), existingEvent.getPlanPhaseName(), effectiveDate);
                 break;
 
-            case CANCEL:
+            case BCD_CHANGE:
+                newEvent = BCDEventData.createBCDEvent(subscription, transferDate, existingEvent.getBillCycleDayLocal());
                 break;
 
+            case CANCEL:
             default:
-                throw new SubscriptionBaseError(String.format("Unexpected transitionType %s", existingEvent.getSubscriptionTransitionType()));
+                break;
+
         }
         return newEvent;
     }
 
     @VisibleForTesting
-    List<SubscriptionBaseEvent> toEvents(final List<ExistingEvent> existingEvents, final DefaultSubscriptionBase subscription,
-                                         final DateTime transferDate, final InternalTenantContext context) throws SubscriptionBaseTransferApiException {
-
-
+    List<SubscriptionBaseEvent> toEvents(final List<ExistingEvent> existingEvents, final DefaultSubscriptionBase subscription, final DateTime transferDate) throws SubscriptionBaseTransferApiException {
         try {
-
             final List<SubscriptionBaseEvent> result = new LinkedList<SubscriptionBaseEvent>();
-
-            SubscriptionBaseEvent event = null;
             ExistingEvent prevEvent = null;
+            ExistingEvent prevBCDEvent = null;
             boolean firstEvent = true;
             for (ExistingEvent cur : existingEvents) {
                 // Skip all events prior to the transferDate
                 if (cur.getEffectiveDate().isBefore(transferDate)) {
-                    prevEvent = cur;
+                    if (cur.getSubscriptionTransitionType() == SubscriptionBaseTransitionType.BCD_CHANGE) {
+                        prevBCDEvent = cur;
+                    } else {
+                        prevEvent = cur;
+                    }
                     continue;
                 }
 
                 // Add previous event the first time if needed
-                if (prevEvent != null) {
-                    event = createEvent(firstEvent, prevEvent, subscription, transferDate, context);
-                    if (event != null) {
-                        result.add(event);
-                        firstEvent = false;
-                    }
+                if (insertEventToResult(firstEvent, prevEvent, subscription, transferDate, result)) {
```

Comment:
I think I mostly managed to convince myself it works but I find the way it's written a tad off / hard to follow 😅 

We're essentially in a `for` loop and the bulk of the work is the last `insertEventToResult` (with the `cur`). The statements before are only there to insert the first `CREATE`/`BCD_CHANGE` events. Since the `if` statements are disjoints, it looks almost as if they could be inserted at any time, even though I believe they need to be insert together as soon as `cur` is after the transfer date.